packages/tangle/tangle.go

```diff
@@ -292,6 +294,7 @@ type Options struct {
 	WeightProvider               WeightProvider
 	SyncTimeWindow               time.Duration
 	StartSynced                  bool
+	CacheTimeProvider            database.CacheTimeProvider
```

Comment:
Use a pointer here. In `Tangle.configure()` assign a default cache time provider (with force time -1):
```go
// Configure modifies the configuration of the Tangle.
func (t *Tangle) Configure(options ...Option) {
	if t.Options == nil {
		t.Options = &Options{
			Store:                        mapdb.NewMapDB(),
			Identity:                     identity.GenerateLocalIdentity(),
			IncreaseMarkersIndexCallback: increaseMarkersIndexCallbackStrategy,
			CacheTimeProvider: database.NewCacheTimeProvider(-1 /* forceCacheTime */),
		}
	}

	for _, option := range options {
		option(t.Options)
	}

	if t.Options.ConsensusMechanism != nil {
		t.Options.ConsensusMechanism.Init(t)
	}
}
```

Reply:
ok,
I deleted the `/* forceCacheTime */` comment because people using IDEs can just read the go docs